pinentryedittext/src/main/java/com/alimuzaffar/lib/pin/PinEntryEditText.java

```diff
@@ -216,16 +220,29 @@ protected void onSizeChanged(int w, int h, int oldw, int oldh) {
             mLastCharPaint.setColor(mOriginalTextColors.getDefaultColor());
             mCharPaint.setColor(mOriginalTextColors.getDefaultColor());
         }
-        int availableWidth = getWidth() - getPaddingRight() - getPaddingLeft();
+        int availableWidth = getWidth() - ViewCompat.getPaddingEnd(this) - ViewCompat.getPaddingStart(this);
         if (mSpace < 0) {
             mCharSize = (availableWidth / (mNumChars * 2 - 1));
         } else {
             mCharSize = (availableWidth - (mSpace * (mNumChars - 1))) / mNumChars;
         }
         mLineCoords = new RectF[(int) mNumChars];
         mCharBottom = new float[(int) mNumChars];
-        int startX = getPaddingLeft();
+        int startX;
         int bottom = getHeight() - getPaddingBottom();
+        int rtlFlag = 1;
+        final boolean isLayoutRtl = TextUtilsCompat.getLayoutDirectionFromLocale(Locale.getDefault()) == ViewCompat.LAYOUT_DIRECTION_RTL;
+        // can't support rtl
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR1) {
```

Comment:
Look at the code for `ViewCompat`, it returns `view.getPaddingLeft()` for everything below JellybeanMR1 so the condition to check for older phones is not needed. `ViewCompat` has a base implementation for `getPaddingStart()` that return `getPaddingLeft()`. This is then overwritten for the JRMR1 implementations where it returns view.getPaddingStart(). So you were correct the first time with this. I recommend removing this if condition and just using the conditions inside the else block.

    if (isLayoutRtl) {
        rtlFlag = -1;
        startX = (int) (getWidth() - ViewCompat.getPaddingStart(this) - mCharSize);
    } else {
        rtlFlag = 1;
        startX = ViewCompat.getPaddingStart(this);
    }